src/docker/composeappengine.cc

```diff
@@ -44,6 +127,19 @@ void ComposeAppEngine::remove(const App& app) {
 }
 
 bool ComposeAppEngine::isRunning(const App& app) const {
+  if (!boost::filesystem::exists(appRoot(app) / MetaDir)) {
```

Comment:
maybe AppState should be able to see this and just return an unknown state?

Reply:
AppState will yield an exception in this case which will lead to the fallback scenario which is checking running Apps through the docker client.

Reply:
Just feels like some API friction here. It seems like you have to know the internals of AppState here to call it correctly? If AppState switched its internals to use `/var/run....` this code becomes invalid?

Reply:
sure it should be just `exists(appRoot(app)` as in `run` and `install`

Reply:
updated